app/src/main/java/org/mozilla/fenix/library/history/HistoryAdapter.kt

```diff
@@ -60,7 +62,16 @@ class HistoryAdapter(
             timeGroup = current.historyTimeGroup
         }
 
-        holder.bind(current, timeGroup, position == 0, mode, isPendingDeletion)
+        holder.bind(
+            current,
+            BindingParams(
```

Comment:
Given my above comment, adding a `BindingParams` data class seems like an unnecessary refactoring here especially since we don't need to pass `syncedHistoryVisible` like this. I would prefer if we just kept it the way it is to avoid too many changes that will overwrite the prior history with unwrapping the parameters within `bind`

Reply:
This change is unnecessary because we don't need to pass this `syncedHistoryVisible` around. We can just call the `FeatureFlags` directly where it is needed.

Reply:
We also suppress plenty of `@Suppress("LongParameterList")` where it makes sense to. `BindingParams` added unnecessary complexity and refactoring when it was not necessary to do so because we were passing `syncedHistoryVisible` around.